packages/typespec-client-generator-core/src/http.ts

```diff
@@ -526,7 +546,7 @@ export function getCorrespondingMethodParams(
       );
       return diagnostics.wrap([]);
     }
-    return diagnostics.wrap([context.__subscriptionIdParameter]);
+    return diagnostics.wrap(subId ? [subId] : []);
   }
 
   // to see if the service parameter is a method parameter or a property of a method parameter
```

Comment:
from this logic, we could not pop a method's model param's property to client, right?

Reply:
yes we can, this is returning corresponding method params, which in this case we're returning a client param